modules/dhcp_common/record.rb

```diff
@@ -5,15 +5,15 @@ module Proxy::DHCP
   # represent a DHCP Record
   class Record
```

Comment:
I think the record class can be removed, it's mostly being used as a container for options hash. Additionally: 
 * subnet should be stored as a string: it's super annoying having to construct a Subnet object every time a lease or resevation is constructed, only so the subnet address string can be used later).
 * Ipv4 and ipv6 can be kept separate, would also allow for meaningful initialization (no set_mac_or_uid-like methods)
* validation should be moved out of constructor (if required at all -- we only create these objects from data received from isc/ms/etc backends)